packages/node-core/src/api.service.ts

```diff
@@ -12,6 +20,21 @@ export abstract class ApiService {
   constructor(protected project: any) {}
 
   abstract init(): Promise<ApiService>;
-
   abstract get api(): any; /*ApiWrapper*/
+
+  async fetchBlocksGeneric<T>(batch: number[], fetchFuncProvider: FetchFunctionProvider<T>, args?: any): Promise<T[]> {
```

Comment:
We don't need args and we should make attempts configurabel

```suggestion
  async fetchBlocksGeneric<T>(fetchFuncProvider: FetchFunctionProvider<T>, batch: number[], numAttempts = MAX_RECONNECT_ATTEMPTS): Promise<T[]> {
```